lib/bokken_web/controllers/skill_controller.ex

```diff
@@ -0,0 +1,125 @@
+defmodule BokkenWeb.SkillController do
+  use BokkenWeb, :controller
+
+  alias Bokken.Curriculum
+  alias Bokken.Curriculum.{MentorSkill, NinjaSkill, Skill}
+
+  action_fallback BokkenWeb.FallbackController
+
+  defguard is_organizer(conn) when conn.assigns.current_user.role === :organizer
+  defguard is_ninja(conn) when conn.assigns.current_user.role === :ninja
+  defguard is_mentor(conn) when conn.assigns.current_user.role === :mentor
+
+  def index(conn, %{"ninja_id" => _ninja_id} = params) do
+    skills = Curriculum.list_ninja_skills(params)
+    render(conn, "index.json", skills: skills)
+  end
+
+  def index(conn, %{"mentor_id" => _mentor_id} = params) do
+    skills = Curriculum.list_mentor_skills(params)
+    render(conn, "index.json", skills: skills)
+  end
+
+  def index(conn, _params) do
+    skills = Curriculum.list_skills()
+    render(conn, "index.json", skills: skills)
+  end
+
+  def create(conn, %{"skill" => skill_params}) when is_organizer(conn) do
+    with {:ok, %Skill{} = skill} <- Curriculum.create_skill(skill_params) do
+      conn
+      |> put_status(:created)
+      |> render("show.json", skill: skill)
+    end
+  end
+
+  def create(conn, %{"skill" => skill_id}) when is_ninja(conn) do
```

Comment:
I don't think a ninja could add its own skills :thinking:

Reply:
They can't add a Skill but they add their NinjaSkill, which is what that does

Reply:
From a ninja's / mentor's point of view, adding a skill means adding it to their skillset, not creating a standalone skill. Only organizers can add skills.

Reply:
Yes, I don't think ninjas could add skills to their skillset. Probably would be safer to let mentors decide which skill each ninja does have. 

I also think mentors could create new skills and not only organizers.

Reply:
But then how do you assign a skill to a first time ninja? How will we know what they want to learn?

Because that will be an issue when we do the pairings later

Reply:
Not to mention that some ninjas can learn skills outside of sessions, which means only mentors be allowed to add skills a little too restricting

Reply:
In the first session, it's always the right time to meet the ninja, present the dojo and find out the best roadmap for the ninja.

This makes me think the name of this resource as far as the ninja is concerned it is not a skillset (that's the badges), this is what he wants/should learn.

Reply:
I agree with you in the sense that this is not a skillset, but what the ninja wants to learn. So, a ninja should be allowed to do it (or it's guardian).